daikon-crypto/crypto-utils/src/main/java/org/talend/daikon/crypto/KeySources.java

```
@@ -124,24 +124,46 @@ public static KeySource systemProperty(String systemProperty) {
+     * PBKDF2 (with Hmac SHA256) digester. It uses an iteration count of "65536".
+     * </p>
+     * <p>
+     * As recommendation, you may initialize a salt using {@link org.talend.daikon.crypto.KeySources#random(int)}.
+     * </p>
+     * 
+     * @deprecated Use {@link #pbkDf2(String, byte[], int, int)} instead with a stronger iteration count.
+     * @return A {@link KeySource} implementation using PBKDF2 and provided <code>salt</code>
+     */
+    @Deprecated
+    public static KeySource pbkDf2(String password, byte[] salt, int keyLength) {
+        return pbkDf2(password, salt, keyLength, 65536);
+    }
+
     /**
      * <p>
      * Returns a {@link KeySource} using the provided password, salt and keyLength values to generate a SecretKey using
      * PBKDF2 (with Hmac SHA256) digester.
      * </p>
      * <p>
      * As recommendation, you may initialize a salt using {@link org.talend.daikon.crypto.KeySources#random(int)}.
+     * The OWASP Password Storage Cheat Sheet recommends an iteration count of 310000 (with HMAC-SHA256)
      * </p>
      *
+     * @param password The password to use to generate the KeySource
+     * @param salt The salt to use
+     * @param keyLength The key length of the generated key in bits
+     * @param iterations The iteration count to use
      * @return A {@link KeySource} implementation using PBKDF2 and provided <code>salt</code>
      */
-    public static KeySource pbkDf2(String password, byte[] salt, int keyLength) {
+    public static KeySource pbkDf2(String password, byte[] salt, int keyLength, int iterations) {
         return () -> {
             if (salt == null || salt.length == 0) {
                 throw new IllegalArgumentException("Cannot use pbkDf2 with empty or null salt.");
             }
             try {
-                KeySpec spec = new PBEKeySpec(password.toCharArray(), salt, 65536, keyLength);
+                KeySpec spec = new PBEKeySpec(password.toCharArray(), salt, iterations, keyLength);
```

Could we throw an exception here if the iterations are really too low, to have a cutoff/threshold/hard requirement in place

---

Yes good idea. Say 50000? No one is using it with less than this anyway at Talend

---

60K and we've got a deal ! 🤝